profiler/profiler.go

```diff
@@ -123,11 +125,18 @@ func newProfiler(opts ...Option) (*profiler, error) {
 	if cfg.uploadTimeout <= 0 {
 		return nil, fmt.Errorf("invalid upload timeout, must be > 0: %s", cfg.uploadTimeout)
 	}
+	for pt := range cfg.types {
+		if _, ok := profileTypes[pt]; !ok {
+			return nil, fmt.Errorf("unknown profile type: %d", pt)
```

Comment:
This approach is perfect. But the error message could be unclear to the user. Technically, it's a _known_ profile type because it's part of the public API and he just used it. This is more like an internal programming error. Maybe something like:
```suggestion
			return nil, fmt.Errorf("profile type internally undefined: %d", pt)
```

Reply:
@gbbr it could be a known `ProfileType`. But it could also be a user passing `ProfileType(123)`. So I think the current error is ok?